Userland/Libraries/LibPthread/pthread.cpp

```diff
@@ -40,6 +41,13 @@ __thread size_t s_stack_size;
 #define __RETURN_PTHREAD_ERROR(rc) \
     return ((rc) < 0 ? -(rc) : 0)
 
+struct CleanupHandler {
+    void (*routine)(void*);
+    void* argument;
+};
+
+static thread_local SinglyLinkedList<CleanupHandler> cleanup_handlers;
```

Comment:
We've had bad times before with C++11 `thread_local` objects in shared objects (rather than the main executable). 

Are we sure that this actually works as intended? The implementation looks more or less correct, but I think it deserves a test (and a sad workaround impl with `__thread` if it turns out it doesn't work)

Reply:
It didn't appear to do anything wrong during the time that I tested the port, but I can specifically test that tomorrow. Do you have any examples on hand about what a "bad time" would be (i.e. what unintended behavior I'm looking for)? What would be the alternative using `__thread`? `SinglyLinkedList<CleanupHandler>*` and an initialization function?

Reply:
@kleinesfilmroellchen Ran into some weird surprises in LibCore with the multithreaded EventLoop refactor.

But... it looks like we're still using thread_local there? Maybe it was entirely due to some bad tests using a static EventLoop or GUI::Application that had the thread_locals trying to be initialized during global initialization? Having a hard time finding the specific behavior...

Reply:
Ah here's the issue. but it went away before the thing got merged in :thonk:

https://github.com/SerenityOS/serenity/issues/11786

Reply:
I think we just ifdef the `thread_local` away or something for DynamicLoader

Reply:
Assuming the issue with `thread_local` was caused by `DynamicLoader` having to do special stuff using `LibC`, is there even any reason to do something here (of course unless I find any immediate issues while testing)? `LibPthread` does not appear to be included when compiling `DynamicLoader`.